packages/frontend/src/lib/Navigation.svelte

```diff
@@ -12,6 +12,11 @@ export let meta: TinroRouteMeta;
 let experimentalTuning: boolean = false;
 let cfgUnsubscribe: Unsubscriber;
 
+// By default, faBookOpen is 576x512, but we want it to be 512x512
+// we cannot modify the width and height in SettingsNavItem so just modify the icon instead
+let faBookOpenIcon = faBookOpen;
```

Comment:
issue: I think you should clone the values of the original object using spread operator

Reply:
Unsure what you mean? Is using 'let' part of cloning it?

Reply:
something like const myIcon = {...originalIcon};

Reply:
but in your case as you've array inside property it won't do a deep clone

So maybe using https://developer.mozilla.org/en-US/docs/Web/API/Window/structuredClone#description method

Reply:
or the hack  JSON.parse(JSON.stringify(object))

Reply:
Hmmm unfortunatley using both:

```
let faBookOpenIcon = JSON.parse(JSON.stringify(faBookOpen))
faBookOpenIcon.icon[0] = 512;
```

or:

```
let faBookOpenIcon = structuredClone(faBookOpen);
```

Actually causes issue. Okay if we just keep the original solution?

We do not modify any of the "original object values" since all are default anyways, we just force 512 width.

Reply:
> We do not modify any of the "original object values" since all are default anyways, we just force 512 width.

You modify the original definition of the icon. The problem is that if I import the same icon in another place, you're changing the value everywhere, not only in Navigation.svelte

I tried
```
let copyFaBookOpenIcon: IconDefinition | undefined = undefined;

onMount(() => {
// By default, faBookOpen is 576x512, but we want it to be 512x512
// we cannot modify the width and height in SettingsNavItem so just modify the icon instead
 copyFaBookOpenIcon = structuredClone(faBookOpen);
 copyFaBookOpenIcon.icon[0] = 512;

```

and then replace occurence of `faBookOpen` by `copyFaBookOpenIcon` and I didn't see issue

what are the issues you're seing ?

Reply:
I was getting errors when it was copying over the JSON. I think I found the issue, I wasn't doing it within onMount! Thanks for the explanation. Never realized it was going to change the value everywhere within svelte.